src/ppl/src/IOPlacer.cpp

```diff
@@ -1338,19 +1338,25 @@ int IOPlacer::assignGroupToSection(const std::vector<int>& io_group,
 
   if (!first_pin.isAssignedToSection() && !first_pin.inFallback()) {
     std::vector<int64_t> dst(sections.size(), 0);
+    std::vector<int64_t> larger_cost(sections.size(), 0);
     for (int i = 0; i < sections.size(); i++) {
       for (int pin_idx : io_group) {
         IOPin& pin = net->getIoPin(pin_idx);
+        bool has_mirrored_pin = pin.getBTerm()->hasMirroredBTerm();
         int pin_hpwl = net->computeIONetHPWL(pin_idx, sections[i].pos);
         if (pin_hpwl == std::numeric_limits<int>::max()) {
           dst[i] = pin_hpwl;
           break;
         }
         dst[i] += pin_hpwl + getMirroredPinCost(pin, sections[i].pos);
+        if (has_mirrored_pin) {
+          larger_cost[i]
+              += std::max(pin_hpwl, getMirroredPinCost(pin, sections[i].pos));
```

Comment:
Are we accumulating here (+=)?  Don't compute getMirroredPinCost twice.

Reply:
Yes, we accumulate because this is for pin groups. We accumulate the total distance of the pin groups, so I decided to accumulate the largest cost between the mirrored pins to have a similar logic.